qiskit/circuit/instruction.py

```diff
@@ -379,3 +380,30 @@ def repeat(self, n):
 
         instruction.definition = [(self, qargs[:], cargs[:])] * n
         return instruction
+
+
+def deprecate_arguments(kwarg_map):
```

Comment:
You didn't want to put this in the extension module somewhere? It doesn't really matter but that would be my first guess on where to put it since it only is used by the modules in the extensions. There or `qiskit/util.py` I guess

Reply:
Good point, moved to `qiskit/util` in 3d5637818acc548fafa9baa7dd323877b1911e23.